academic_observatory_workflows/workflows/unpaywall_data_feed_telescope.py

```diff
@@ -0,0 +1,326 @@
+    ) -> Generator:
+        """Calculate the scheduled days apart.
+
+        :param start_date: DAG start date.
+        :param schedule_interval: DAG schedule interval.
+        :return: A generator that gives back the days apart for each execution.
+        """
+
+        if isinstance(schedule_interval, (timedelta, relativedelta)):
+            while True:
+                yield schedule_interval.days
+
+        a = start_date
+        it = croniter(schedule_interval, start_date)
+
+        while True:
+            b = it.next(datetime)
+            diff = (b - a).days
+            a = b
+            yield diff
+
+    def validate_schedule_interval_(self, *, start_date: pendulum.DateTime, schedule_interval: str):
+        """Check that the schedule interval gives us 1 or 7 day differences.
+        Throws exception on failure.
+
+        :param start_date: DAG start date.
+        :param schedule_interval: DAG schedule interval.
+        """
+
+        days_apart = UnpaywallDataFeedTelescope.schedule_days_apart_(
+            start_date=start_date, schedule_interval=schedule_interval
+        )
+
+        diffs = [next(days_apart) for i in range(2)]
+        if diffs[0] != diffs[1] or diffs[0] != 1:
+            raise AirflowException(f"Schedule interval must trigger executions 1 days apart.")
+
+    def check_releases(self, **kwargs) -> bool:
+        """Check to see if diff releases are available. If not, and it's not the first release, then skip doing work.
+        Snapshot releases are checked on first release at download stage.
+
+        :param kwargs: The context passed from the PythonOperator.
+        :return: True to continue, False to skip.
+        """
+
+        start_date, first_release = self.get_release_info_(**kwargs)
+
+        # No checks on first release
+        if first_release:
+            return True
+
+        # Check for diffs
+        api_key = get_airflow_connection_password(UnpaywallDataFeedRelease.AIRFLOW_CONNECTION)
+        url = f"{UnpaywallDataFeedRelease.CHANGEFILES_URL}?interval=day&api_key={api_key}"
+        _, filename = UnpaywallDataFeedRelease.get_diff_release(feed_url=url, start_date=start_date)
+
+        # No release within our target date.
+        if filename is None:
+            return False
+
+        # Release found
+        return True
+
+    def make_release(self, **kwargs) -> UnpaywallDataFeedRelease:
+        """Make a Release instance
+
+        :param kwargs: The context passed from the PythonOperator.
+        :return: UnpaywallDataFeedRelease
+        """
+
+        start_date, first_release = self.get_release_info_(**kwargs)
+        release = UnpaywallDataFeedRelease(
+            dag_id=self.dag_id,
+            start_date=start_date,
+            end_date=start_date,
+            first_release=first_release,
+        )
+        return release
+
+    def get_release_info_(self, **kwargs) -> Tuple[pendulum.DateTime, bool]:
```

Comment:
Private methods in Python should begin with an underscore, not end with an underscore, e.g. _download_snapshot.